src/tree/registries/UnifiedRegistryTreeDataProvider.ts

```diff
@@ -132,19 +132,46 @@ export class UnifiedRegistryTreeDataProvider implements vscode.TreeDataProvider<
         void this.refresh();
     }
 
-    public async getAllConnectedRegistries(): Promise<UnifiedRegistryItem<CommonRegistry>[]> {
-        const registryRoots = await this.getChildren();
+    /**
+     *
+     * @param imageBaseName The base name of the image to find registries for. e.g. 'docker.io'
+     * @returns A list of registries that are connected to the extension. If imageBaseName is provided, only registries that
+     *          can be used to push to that image will be returned.
+     */
+    public async getConnectedRegistries(imageBaseName?: vscode.Uri): Promise<UnifiedRegistryItem<CommonRegistry>[]> {
+        let registryRoots = await this.getChildren();
+        let findAzureRegistryOnly = false;
+
+        // filter out registry roots that don't match the image base name
+        if (imageBaseName) {
+            const authority = imageBaseName.authority;
+
+            if (authority === 'docker.io') {
+                registryRoots = registryRoots.filter(r => (r.wrappedItem as CommonRegistryRoot).label === 'Docker Hub');
+            }
+            else if (authority.endsWith('azurecr.io')) {
+                registryRoots = registryRoots.filter(r => (r.wrappedItem as CommonRegistryRoot).label === 'Azure');
+                findAzureRegistryOnly = true;
+            }
+            else {

```

Comment:
You could also cover GitHub here (ghcr.io)

Reply:
good point